docs_src/microservices/device/supported/device-onvif-camera/Walkthrough/deployment.md

```diff
@@ -241,72 +253,155 @@ Follow these instructions to update devices.
-      Response [200] true
-
-
-
-      Secret Name: a
-      curl -X GET http://localhost:8500/v1/kv/edgex/v3/device-onvif-camera/AppCustom/CredentialsMap/a?raw=true
-      Response [200] 
-      Setting Credentials Map: a = '11:22:33:44:55:66'
-      curl --data "11:22:33:44:55:66" -X PUT http://localhost:8500/v1/kv/edgex/v3/device-onvif-camera/AppCustom/CredentialsMap/a
-      Response [200] true
-      ``` 
+    === "Secure mode"
+        !!! note
+            If running in secure mode all the api executions need the [JWT token](#token-generation) generated previously.
+
+        
+        a. Enter your chosen credentials name and then execute the command to set the mapping name.
+
+        !!! note
+            To edit credentials, this command may be skipped, and then use one of the existing credential mappings in the following commands.
+
+        ```bash
+        curl --data '<creds-name>' -H "X-Consul-Token:<consul-token>" -X PUT "http://localhost:8500/v1/kv/edgex/v3/device-onvif-camera/Writable/InsecureSecrets/credentials001/SecretName"
+        ```
+        
+        b. Enter your chosen credentials name and then execute the command to create the mapping.
+        ```bash
+        curl -H "X-Consul-Token:<consul-token>" -X PUT "http://localhost:8500/v1/kv/edgex/v3/device-onvif-camera/AppCustom/CredentialsMap/<creds-name>"
+        ```
+
+        c. Enter your chosen username, password, and authentication mode and credentials name and then execute the command to create the mapping.
+        !!! note
+            The options for authentication method are: `usernametoken`, `digest`, or `both`
+
+        ```bash
+        curl --data '{
+                    "apiVersion":"v3",
+                    "secretName": "<creds-name>",
+                    "secretData":[
+                        {
+                            "key":"username",
+                            "value":"<username>"
+                        },
+                        {
+                            "key":"password",
+                            "value":"<password>"
+                        },
+                        {
+                            "key":"mode",
+                            "value":"<auth-mode>"
+                        }
+                    ]
+                }' --header 'Authorization:Bearer <JWT-token>' -X POST "http://localhost:59984/api/v3/secret"
+        ```
+
+        d. Enter your mac-address(es) and then execute the command to add the mac address(es) to the mapping.
+        !!! note
+            If you want to map multiple mac addresses, enter a comma separated list in the command
+
+        ```bash
+        curl --data '<mac-address>' -H "X-Consul-Token:<consul-token>" -X PUT "http://localhost:8500/v1/kv/edgex/v3/device-onvif-camera/AppCustom/CredentialsMap/<creds-name>"
+        ```
+        
+        e. Check the status of the credentials map.
```

Comment:
From what I can tell it needs some kind of query to return anything.

Reply:
Ok yeah. it looks like you can get it to print the info with `?recurse=true`, but it will print some base64 values and other stuff:

```json
 curl -H "X-Consul-Token:<consul-token>" -X GET "http://localhost:8500/v1/kv/edgex/v3/device-onvif-camera/AppCustom/CredentialsMap?recurse=true" | jq .         
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   319  100   319    0     0   535k      0 --:--:-- --:--:-- --:--:--  311k
[
  {
    "LockIndex": 0,
    "Key": "edgex/v3/device-onvif-camera/AppCustom/CredentialsMap/NoAuth",
    "Flags": 0,
    "Value": null,
    "CreateIndex": 211,
    "ModifyIndex": 211
  },
  {
    "LockIndex": 0,
    "Key": "edgex/v3/device-onvif-camera/AppCustom/CredentialsMap/credentials001",
    "Flags": 0,
    "Value": "MTI6MzQ6NTY6Nzg6OTA6MTM=",
    "CreateIndex": 267,
    "ModifyIndex": 312
  }
]

```